### PR TITLE
Posting a question

### PR DESCRIPTION
**Pull request by mshakhmaykin**
Hey, could you please take a look at the issue we're having in our pipeline? Maybe you'll get an idea how to solve it?

TL;DR:
It runs tests in Cypress.
It captures reports and stores them in artifacts.
It finds and publishes HTMLreport.

But HTMLViewer opens blank page, because it refers to external `assets/` instead of opening them from our report/index.html

<details><summary>Job definitions</summary>

```
steps:
  # Run Cypress with config parameters, additional arguments, and pipelines reporter
  - task: Yarn@2
    displayName: '${{ parameters.project }}: ${{ parameters.env }}'
    condition: succeededOrFailed()
    inputs:
      ProjectDirectory: projects/${{ parameters.project }}
      Arguments: run:${{ parameters.env }} ${{ parameters.additionalArguments }}

  - task: PublishBuildArtifacts@1
    displayName: Publish Artifacts
    condition: succeededOrFailed()
    inputs:
      pathToPublish: '$(System.DefaultWorkingDirectory)/projects/${{ parameters.project }}/cypress/reports/html'
      artifactName: HTMLreportData

  # Publish Test Results (Pipelines reporter)
  - task: PublishTestResults@2
    displayName: Publish Test Results
    condition: succeededOrFailed()
    inputs:
      testResultsFiles: $(System.DefaultWorkingDirectory)/projects/${{ parameters.project }}/cypress/reports/junit/*.xml
      searchFolder: ''
      mergeTestResults: true
      testRunTitle: ${{ parameters.project }} - ${{ parameters.env }}

  - task: PublishHtmlReport@1
    displayName: 'Publish HTML Report'
    condition: succeededOrFailed()
    inputs:
      #reportDir: $(System.DefaultWorkingDirectory)/projects/${{ parameters.project }}/cypress/reports/html/index.html
      reportDir: 'projects/${{ parameters.project }}/cypress/reports/html/index.html'
```

</details>

<details><summary>Debug log of relevant jobs</summary>

```
##[debug]Evaluating condition for step: 'Publish Artifacts'
##[debug]Evaluating: succeededOrFailed()
##[debug]Evaluating succeededOrFailed:
##[debug]=> True
##[debug]Result: True
Starting: Publish Artifacts
==============================================================================
Task         : Publish build artifacts
Description  : Publish build artifacts to Azure Pipelines or a Windows file share
Version      : 1.229.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/publish-build-artifacts
==============================================================================
##[debug]Using node path: /home/vsts/agents/3.227.2/externals/node16/bin/node
##[debug]agent.TempDirectory=/home/vsts/work/_temp
##[debug]loading inputs and endpoints
##[debug]loading INPUT_PATHTOPUBLISH
##[debug]loading INPUT_ARTIFACTNAME
##[debug]loading INPUT_ARTIFACTTYPE
##[debug]loading INPUT_MAXARTIFACTSIZE
##[debug]loading INPUT_PARALLEL
##[debug]loading INPUT_PARALLELCOUNT
##[debug]loading INPUT_STOREASTAR
##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
##[debug]loading SECRET_SYSTEM_ACCESSTOKEN
##[debug]loaded 11
##[debug]Agent.ProxyUrl=undefined
##[debug]Agent.CAInfo=undefined
##[debug]Agent.ClientCert=undefined
##[debug]Agent.SkipCertValidation=undefined
##[debug]check path : /home/vsts/work/_tasks/PublishBuildArtifacts_2ff763a7-ce83-4e1f-bc89-0ae63477cebe/1.229.0/task.json
##[debug]adding resource file: /home/vsts/work/_tasks/PublishBuildArtifacts_2ff763a7-ce83-4e1f-bc89-0ae63477cebe/1.229.0/task.json
##[debug]system.culture=en-US
##[debug]StoreAsTar=false
##[debug]PathtoPublish=/home/vsts/work/1/s/projects/Compliance-Management/cypress/reports/html
##[debug]check path : /home/vsts/work/1/s/projects/Compliance-Management/cypress/reports/html
##[debug]ArtifactName=HTMLreportData
##[debug]ArtifactType=Container
##[debug]system.hostType=build
##[debug]MaxArtifactSize=0


---------------------------------------------------------------------------------------------------------



##[debug]Evaluating condition for step: 'Publish Test Results'
##[debug]Evaluating: succeededOrFailed()
##[debug]Evaluating succeededOrFailed:
##[debug]=> True
##[debug]Result: True
Starting: Publish Test Results
==============================================================================
Task         : Publish Test Results
Description  : Publish test results to Azure Pipelines
Version      : 2.229.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/test/publish-test-results
==============================================================================
##[debug]Using node path: /home/vsts/agents/3.227.2/externals/node16/bin/node
##[debug]agent.TempDirectory=/home/vsts/work/_temp
##[debug]loading inputs and endpoints
##[debug]loading INPUT_TESTRUNNER
##[debug]loading INPUT_TESTRESULTSFILES
##[debug]loading INPUT_MERGETESTRESULTS
##[debug]loading INPUT_FAILTASKONFAILEDTESTS
##[debug]loading INPUT_FAILTASKONMISSINGRESULTSFILE
##[debug]loading INPUT_TESTRUNTITLE
##[debug]loading INPUT_PUBLISHRUNATTACHMENTS
##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
##[debug]loading SECRET_SYSTEM_ACCESSTOKEN
##[debug]loaded 11
##[debug]Agent.ProxyUrl=undefined
##[debug]Agent.CAInfo=undefined
##[debug]Agent.ClientCert=undefined
##[debug]Agent.SkipCertValidation=undefined
##[debug]check path : /home/vsts/work/_tasks/PublishTestResults_0b0f01ed-7dde-43ff-9cbb-e48954daf9b1/2.229.1/task.json
##[debug]adding resource file: /home/vsts/work/_tasks/PublishTestResults_0b0f01ed-7dde-43ff-9cbb-e48954daf9b1/2.229.1/task.json
##[debug]system.culture=en-US
##[debug]testRunner=JUnit
##[debug]testResultsFiles=/home/vsts/work/1/s/projects/Compliance-Management/cypress/reports/junit/*.xml
##[debug]mergeTestResults=true
##[debug]platform=undefined
##[debug]configuration=undefined
##[debug]testRunTitle=Compliance-Management - dev1
##[debug]publishRunAttachments=true



----------------------------------------------------------------------------------------------------------


##[debug]Evaluating condition for step: 'Publish HTML Report'
##[debug]Evaluating: succeededOrFailed()
##[debug]Evaluating succeededOrFailed:
##[debug]=> True
##[debug]Result: True
Starting: Publish HTML Report
==============================================================================
Task         : Publish Html Report
Description  : Publish Html Report
Version      : 1.0.14
Author       : 
Help         : Replace with markdown to show in help
==============================================================================
##[debug]Using node path: /home/vsts/agents/3.227.2/externals/node10/bin/node
##[debug]agent.TempDirectory=/home/vsts/work/_temp
##[debug]loading inputs and endpoints
##[debug]loading INPUT_TABNAME
##[debug]loading INPUT_REPORTDIR
##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
##[debug]loading SECRET_SYSTEM_ACCESSTOKEN
##[debug]loaded 6
##[debug]Agent.ProxyUrl=undefined
##[debug]Agent.CAInfo=undefined
##[debug]Agent.ClientCert=undefined
##[debug]Agent.SkipCertValidation=undefined
##[debug]reportDir=/home/vsts/work/1/s/projects/Compliance-Management/cypress/reports/html/index.html
##[debug]check path : /home/vsts/work/1/s/projects/Compliance-Management/cypress/reports/html/index.html
##[debug]Agent.JobName=Run Compliance-Management Tests
##[debug]System.StageDisplayName=__default
##[debug]System.StageAttempt=1
##[debug]tabName=..myTab
/home/vsts/work/1/s/projects/Compliance-Management/cypress/reports/html/index.html
##[debug]Processed: ##vso[task.addattachment type=report-html;name=..myTab.run-compliance-management-tests.__default.1;]/home/vsts/work/1/s/projects/Compliance-Management/cypress/reports/html/index.html
Finishing: Publish HTML Report
```

</details>

<details><summary>Screenshots</summary>

BlankPage

Artifacts bundle

</details>

Both 404-s above are because of the missing files:
https://awardedsolutions.gallerycdn.vsassets.io/extensions/awardedsolutions/azure-pipelines-html-report-awardedsolutions/1.0.16/1684005167596/dist/assets/app.js
https://awardedsolutions.gallerycdn.vsassets.io/extensions/awardedsolutions/azure-pipelines-html-report-awardedsolutions/1.0.16/1684005167596/dist/assets/app.css

### CODE REVIEW

**Comment by aurel4oxand** [comment]
> Hey, could you please take a look at the issue we're having in our pipeline? Maybe you'll get an idea how to solve it?
> 
> TL;DR: It runs tests in Cypress. It captures reports and stores them in artifacts. It finds and publishes HTMLreport.
> 
> But HTMLViewer opens blank page, because it refers to external `assets/` instead of opening them from our report/index.html
> 
> Job definitions
> Debug log of relevant jobs
> Screenshots
> Both 404-s above are because of the missing files: https://awardedsolutions.gallerycdn.vsassets.io/extensions/awardedsolutions/azure-pipelines-html-report-awardedsolutions/1.0.16/1684005167596/dist/assets/app.js https://awardedsolutions.gallerycdn.vsassets.io/extensions/awardedsolutions/azure-pipelines-html-report-awardedsolutions/1.0.16/1684005167596/dist/assets/app.css

Hi @mshakhmaykin 

I ran into similar issue, and I found a workaround which provides good results

(EDIT : Thanks to [this post](https://medium.com/red-buffer/cypress-generating-mochawesome-report-42d6436eb6f3) that put me on the right track 👍 )

I still don't know why assets are not loaded properly but using `inlineAssets` option with `cypress-mochawesome-reporter` did the trick (probably Azure DevOps restrictions to reach external resources ?)

Anyway, adding following `reporterOptions` to your `cypress.config.js` file produces a local HTML file with embedded styles and scripts : 

```
module.exports = defineConfig({
  reporter: 'cypress-mochawesome-reporter',
  reporterOptions: {
    charts: true,
    inlineAssets: true,
    embeddedScreenshots: true
  },
  ...
```

Here is the result : 

image

Hope this helps 😉
